### PR TITLE
Mangastream Update:

### PR DESCRIPTION
**Pull request by daarkdemon**
- add generic selectors
- add 2nd language support for single source
- fix search(title with whitespaces)
- fix default viewer for Manga(Rtl)
- fix chapter title
- fix chapter numbers
- optimize helper fuctions
- add Flamescans(English & Arabic)
- add Acescans(English)
- add Kraw(English & Korean)
- add Infernalvoidscans(English)
- add Luminousscans(English)
- add Realmscans(English)
- add Sushiscan(French)
- add Aresmanga(Arabic)
- add Mangasol(Arabic)
- add Swatmanga(Arabic)

Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [ ] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device

### CODE REVIEW

**Review by beerpiss — CHANGES_REQUESTED** [review]
[src/rust/mangastream/sources/acescans/Icon.png](https://github.com/Skittyblock/aidoku-community-sources/pull/51/files#diff-22e2d2f2e309e6e050140f38a430e6944da6e746cdf870ccf56b8da9cc2e5cfc) probably isn't needed.

File: src/rust/mangastream/sources/asurascans/res/filters.json [hunk 1]
```diff
@@ -290,6 +290,7 @@
                 "type": "genre",
                 "name": "Video Games",
                 "id": "video-games"
+
```
Comment (beerpiss) [inline src/rust/mangastream/sources/asurascans/res/filters.json:293]: delete this newline

File: src/rust/mangastream/sources/flamescans/src/helper.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		"Time Travel" => "211",
+		"Tragedy" => "64",
+		"Transmigration" => "185",
+		"Video Games" => "29",
+		"VR" => "162",
+		"Zombies" => "68",
+		_ => "",
+	};
+	String::from(id)
+}
```
Comment (beerpiss) [inline src/rust/mangastream/sources/flamescans/src/helper.rs:83]: Can't you get filter IDs from `filter.object.get("id")` for genre filters?

**Review by beerpiss — CHANGES_REQUESTED** [review]
Add final newlines to your files and run `cargo fmt`/`cargo clippy` on the code.

I hope you've tested all of them because I don't think I have the energy to review 70 files, god bless Skitty

File: src/rust/mangastream/sources/sushiscan/res/filters.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        "name": "Type",
+        "options": [
+            "All",
+            "Manga",
+            "Manhwa",
+            "Manhua",
+            "Comic"
+        ]
+    }
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/sushiscan/res/filters.json:177]: newline

File: src/rust/mangastream/sources/sushiscan/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#![no_std]
+use aidoku::{
+	prelude::*, error::Result, std::{String}, std::Vec, std::net::Request,
```
Comment (beerpiss) [inline src/rust/mangastream/sources/sushiscan/src/lib.rs:3]:
```suggestion
	prelude::*, error::Result, std::String, std::Vec, std::net::Request,
```

**Review by beerpiss — CHANGES_REQUESTED** [review]
- there's an Icon.bak.png in `luminousscans`, also the icon is not a square.
- please insert your final newlines

**Review by beerpiss — CHANGES_REQUESTED** [review]
Bump ReadKomik to v2

**Comment by beerpiss** [comment]
You may also want to add Kiryuu from #52 since their PR won't be usable when you update the template.

File: src/rust/mangastream/sources/asurascans/src/helper.rs [hunk 5]
```diff
@@ -16,5 +17,80 @@ pub fn get_lang_code() -> String {
...
+		"Villain" => "villain",
+		"Virtual Game" => "virtual-game",
+		"Virtual Reality" => "sanal-gerceklik",
+		"Virtual World" => "vr",
+		"Webtoon" => "webtoon",
+		"Wuxia" => "wuxia",
+		_ => "",
+	};
+	String::from(id)
+}
```
[the remark this answers is not in the document]
Comment (daarkdemon) [reply src/rust/mangastream/sources/asurascans/src/helper.rs:96]: This filter ids are for turkish language genres as they are different so i tried to match the same text that are in english, as w le don't have dynamic filtering

File: src/rust/mangastream/sources/blackarmy/src/lib.rs [hunk 6]
```diff
@@ -1,44 +1,55 @@
 #![no_std]
 use aidoku::{
-	prelude::*, error::Result, std::String, std::Vec, std::net::Request,
+	prelude::*, error::Result, std::{String}, std::Vec, std::net::Request,
```
[the remark this answers is not in the document]
Comment (daarkdemon) [reply src/rust/mangastream/sources/blackarmy/src/lib.rs:3]: I may have used another std lib and then remove so forgot to remove it

File: src/rust/mangastream/sources/flamescans/src/helper.rs (hunk 2 above)
Comment (daarkdemon) [reply src/rust/mangastream/sources/flamescans/src/helper.rs:83]: I have mapped the en language tags to turkish tags as the source is multilingual

File: src/rust/mangastream/sources/alphascans/res/filters.json [hunk 7]
```diff
@@ -149,4 +149,4 @@
             "Comic"
         ]
     }
-]
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/alphascans/res/filters.json:152]: add a final newline

File: src/rust/mangastream/sources/asurascans/res/filters.json [hunk 8]
```diff
@@ -344,4 +345,4 @@
             "Comic"
         ]
     }
-]
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/asurascans/res/filters.json:348]: add newlines not remove them

File: src/rust/mangastream/sources/blackarmy/res/filters.json [hunk 9]
```diff
@@ -214,4 +214,4 @@
             "Comic"
         ]
     }
-]
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/blackarmy/res/filters.json:217]: add newlines

File: src/rust/mangastream/sources/kumascans/res/filters.json [hunk 10]
```diff
@@ -424,4 +424,4 @@
             "Comic"
         ]
     }
-]
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/kumascans/res/filters.json:427]: insert a final newline

File: src/rust/mangastream/sources/readkomik/res/filters.json [hunk 11]
```diff
@@ -334,4 +334,4 @@
             "Comic"
         ]
     }
-]
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/readkomik/res/filters.json:337]: newlines newlines

File: src/rust/mangastream/sources/voidscans/res/filters.json [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+        "name": "Type",
+        "options": [
+            "All",
+            "Manga",
+            "Manhwa",
+            "Manhua",
+            "Comic"
+        ]
+    }
+]
```
Comment (beerpiss) [inline src/rust/mangastream/sources/voidscans/res/filters.json:202]: newlines

**Review by beerpiss — CHANGES_REQUESTED** [review]
You still haven't bumped ReadKomik's version.

**Review by beerpiss — APPROVED** [review]

**Pull request merged**